### PR TITLE
Cppcheck

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: src/database/sql_database.cc [hunk 1]
```diff
@@ -595,11 +595,12 @@ std::shared_ptr<CdsObject> SQLDatabase::loadObject(int objectID)
 
     beginTransaction("loadObject");
     auto res = select(qb);
-    std::unique_ptr<SQLRow> row;
-    if (res != nullptr && (row = res->nextRow()) != nullptr) {
-        auto result = createObjectFromRow(row);
-        commit("loadObject");
-        return result;
+    if (res != nullptr) {
+        if (auto row = res->nextRow(); row != nullptr) {
```
Comment (KarlStraussberger) [inline src/database/sql_database.cc:599]: Separate statements

File: src/database/sql_database.cc [hunk 2]
```diff
@@ -612,11 +613,12 @@ std::shared_ptr<CdsObject> SQLDatabase::loadObjectByServiceID(const std::string&
     qb << sql_browse_query << " WHERE " << TQBM(BrowseCol::service_id) << '=' << quote(serviceID);
     beginTransaction("loadObjectByServiceID");
     auto res = select(qb);
-    std::unique_ptr<SQLRow> row;
-    if (res != nullptr && (row = res->nextRow()) != nullptr) {
-        auto result = createObjectFromRow(row);
-        commit("loadObjectByServiceID");
-        return result;
+    if (res != nullptr) {
+        if (auto row = res->nextRow(); row != nullptr) {
```
Comment (KarlStraussberger) [inline src/database/sql_database.cc:617]: Separate statements

File: src/database/sql_database.cc [hunk 3]
```diff
@@ -779,8 +781,7 @@ std::vector<std::shared_ptr<CdsObject>> SQLDatabase::search(const std::unique_pt
     beginTransaction("search");
     auto sqlResult = select(countSQL);
     commit("search");
-    std::unique_ptr<SQLRow> countRow = sqlResult->nextRow();
-    if (countRow != nullptr) {
+    if (auto countRow = sqlResult->nextRow(); countRow != nullptr) {
```
Comment (KarlStraussberger) [inline src/database/sql_database.cc:784]: Sepatate

File: src/database/sql_database.cc [hunk 4]
```diff
@@ -845,10 +846,11 @@ int SQLDatabase::getChildCount(int contId, bool containers, bool items, bool hid
     auto res = select(qb);
     commit("getChildCount");
 
-    std::unique_ptr<SQLRow> row;
-    if (res != nullptr && (row = res->nextRow()) != nullptr) {
-        int childCount = std::stoi(row->col(0));
-        return childCount;
+    if (res != nullptr) {
+        if (auto row = res->nextRow(); row != nullptr) {
```
Comment (KarlStraussberger) [inline src/database/sql_database.cc:850]: Separate

File: src/database/sql_database.cc [hunk 5]
```diff
@@ -1044,8 +1046,7 @@ void SQLDatabase::addContainerChain(std::string virtualPath, const std::string&
     beginTransaction("addContainerChain");
     auto res = select(qb);
     if (res != nullptr) {
-        std::unique_ptr<SQLRow> row = res->nextRow();
-        if (row != nullptr) {
+        if (auto row = res->nextRow(); row != nullptr) {
```
Comment (KarlStraussberger) [inline src/database/sql_database.cc:1049]: Separate

File: src/database/sql_database.cc [hunk 6]
```diff
@@ -1875,8 +1876,7 @@ void SQLDatabase::updateAutoscanList(ScanMode scanmode, std::shared_ptr<Autoscan
         }
         commit("updateAutoscanList x");
 
-        std::unique_ptr<SQLRow> row;
-        if ((row = res->nextRow()) != nullptr) {
+        if (auto row = res->nextRow(); row != nullptr) {
```
Comment (KarlStraussberger) [inline src/database/sql_database.cc:1879]: Separate

File: src/database/sql_database.cc [hunk 7]
```diff
@@ -2086,8 +2086,7 @@ int SQLDatabase::_getAutoscanObjectID(int autoscanID)
     auto res = select(q);
     if (res == nullptr)
         throw DatabaseException("", "error while doing select on ");
-    std::unique_ptr<SQLRow> row;
-    if ((row = res->nextRow()) != nullptr && !row->col(0).empty())
+    if (auto row = res->nextRow(); row != nullptr && !row->col(0).empty())
```
Comment (KarlStraussberger) [inline src/database/sql_database.cc:2089]: Separate

**Review by KarlStraussberger — CHANGES_REQUESTED** [review]
Please do not merge statements in if. That becomes unreadable.

**Comment by neheb** [comment]
Done.

**Review by KarlStraussberger — APPROVED** [review]

File: src/content/content_manager.cc [hunk 8]
```diff
@@ -81,6 +81,12 @@ ContentManager::ContentManager(const std::shared_ptr<Context>& context,
     , session_manager(context->getSessionManager())
     , context(context)
     , timer(std::move(timer))
+#ifdef HAVE_JS
```
Comment (whyman) [inline src/content/content_manager.cc:84]: Why?

File: src/database/database.cc [hunk 9]
```diff
@@ -44,20 +44,18 @@ Database::Database(std::shared_ptr<Config> config)
 std::shared_ptr<Database> Database::createInstance(const std::shared_ptr<Config>& config, const std::shared_ptr<Timer>& timer)
 {
     std::string type = config->getOption(CFG_SERVER_STORAGE_DRIVER);
-    std::shared_ptr<Database> database;
-
-    if (type == "sqlite3") {
-        database = std::static_pointer_cast<Database>(std::make_shared<Sqlite3Database>(config, timer));
-    }
+    auto database = [&] {
```
Comment (whyman) [inline src/database/database.cc:47]: Personally I think the old code is clearer

**Comment by KarlStraussberger** [comment]
@whyman do you have any idea about the sourcehut issue?

**Comment by KarlStraussberger** [comment]
> 
> 
> Hmm, not sure. They were talking about requiring payment, so I have put some money in the pot, hopefully that helps!

It helped. Thank you. We should add a donations button somewhere to cover your expenses.

File: src/content/content_manager.cc (hunk 8 above)
Comment (neheb) [reply src/content/content_manager.cc:84]: It's preferred to initialize in the list instead of the body.

File: src/database/database.cc (hunk 9 above)
Comment (neheb) [reply src/database/database.cc:47]: The goal here is to avoid database ever being NULL as that results in warnings.

Comment (neheb) [reply src/database/database.cc]: I redid this slightly.

**Comment by neheb** [comment]
I'm not sure of the last commit. Comments welcome.

File: src/content/layout/js_layout.h [hunk 10]
```diff
@@ -34,6 +34,7 @@
 
 #include <memory>
 
+#include "content/scripting/import_script.h"
```
Comment (KarlStraussberger) [inline src/content/layout/js_layout.h:37]: If you add the include, please adjust the forward declarations.

**Review by KarlStraussberger — CHANGES_REQUESTED** [review]
For the future: please separate large commits, it's hard to validate the PR.

File: src/content/layout/js_layout.h (hunk 10 above)
Comment (neheb) [reply src/content/layout/js_layout.h:37]: Removed all of them as import_script.h has them.

The actual reason for this change was the destructor removal. make_unique needs a declaration of ImportScript

**Review by KarlStraussberger — APPROVED** [review]

**Pull request merged**